fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/FedoraEventImpl.java

```diff
@@ -255,5 +253,15 @@ public static FedoraEvent from(final Event event) {
         }
     }
 
+    private static String cleanPath(final Event event) throws RepositoryException {
+        // remove any trailing data for property changes
+        final String path = PROPERTY_TYPES.contains(event.getType()) ?
+            event.getPath().substring(0, event.getPath().lastIndexOf("/")) : event.getPath();
+
+        // reformat any hash URIs and remove any trailing /jcr:content
+        final HashConverter converter = new HashConverter();
```

Comment:
Is `HashConverter` stateful? If not, can this be a `static` field, instead of building a new object over and over? The compiler/runtime might be smart enough to factor this out, but we should help it as much as possible.